### ISSUE TITLE
Intellibrite "sneaks" in to dashPanel

### ISSUE THREAD

**Issue by arrmo** [rstrouse/nodejs-poolController-dashPanel#16]
OK, this will seem odd I know, but ... if I leave dashPanel running for a bit, after some period of time I see Intellibrite (that I don't have) - pic below.
![image](https://user-images.githubusercontent.com/8756384/104129196-58029800-5331-11eb-9bbf-7f8042b8ae59.png)

If I refresh, it goes away (again, pic) ... but returns after a period of time it seems. Or am I just nuts? 😆.
![image](https://user-images.githubusercontent.com/8756384/104129197-5fc23c80-5331-11eb-9e6d-4794bcb33318.png)

PS: sometimes it seems like a refresh makes "Loading" occur again (see the pic above) - just to understand, what does that mean?

Thanks!

**Comment by rstrouse** [comment]
Grab the next repo of njsPC and update dashPanel.  We have been working on Touch integration in dashPanel.

**Comment by rstrouse** [comment]
When it says loading it means that njspc is interrogating your OCP for any changes.  Touch interfaces will do this periodically to make sure it is up to date with the anything that may not have been reported or got jumbled up on the RS485 bus.

**Comment by arrmo** [comment]
> Grab the next repo of njsPC and update dashPanel. We have been working on Touch integration in dashPanel.

Pulled the latest (next branch for njsPC, master for dashPanel ... right?). Seems to behave the same? I did restart both of them (using PM2).

Thanks!

**Comment by rstrouse** [comment]
Please hard refresh your browser that lone palette on the lights header should not exist.  I'll bet it is still running the cached files on the client.

**Comment by arrmo** [comment]
NP! Did a hard browser refresh (Ctrl-F5) ... it shows that item (Intellibrite) at ~ 70% of loading, then goes away. But it comes back.

Thanks!

**Comment by arrmo** [comment]
BTW, thinking out loud, but ... I had a discussion with @tagyoureit before, and while getting influxDB working I had to do the following (to stop IntelliBrite from showing up in influxDB),
```
diff --git a/web/bindings/influxDB.json b/web/bindings/influxDB.json
index 030e62a..0d56477 100644
--- a/web/bindings/influxDB.json
+++ b/web/bindings/influxDB.json
@@ -658,6 +658,7 @@
     {
       "name": "lightGroup",
       "description": "Bind lightGroup emit",
+      "enabled": false,
       "points": [
         {
           "measurement": "circuits",
```

So perhaps IntelliBrite is "coming from" njsPC somehow, and you are just displaying it?

**Comment by rstrouse** [comment]
Yes it is simply picking up its existence.  I believe it just gets added no matter what because, If you have a Touch controller it is part of the motherboard so it always exists with no way to disable it.  I'll have a look to see whether there is a way to simply remove it in njspc when there are no lights attached to it.

**Comment by arrmo** [comment]
That makes sense, thanks! I'm not so sure this device even exists in my (~ 17 year old) MB, but I may be wrong 😜.

**Comment by rstrouse** [comment]
It exists. There are cave drawings in southern Arizona to prove it.

**Comment by rstrouse** [comment]
By chance do you have any circuits that have a function defined as any of the following:
SAM Light
SAL Light
Photon Gen
Color Wheel
IntelliBrite
MagicStream


Also if you have that palette showing up on the lights header you don't have the most recent software or it is still running on cached files for dashPanel.

**Comment by arrmo** [comment]
Don't think I have any of those defined ... LOL. Check config.json I assume?

Hmmm ... I do have the palette (you mean the pic below, right?). Odd, CTRL-F5 is supposed force a hard refresh, no? It did force reload (dashPanel ... counting up, then Ready).
![image](https://user-images.githubusercontent.com/8756384/105639064-c2403000-5e3b-11eb-93b4-0349ee465647.png)

Let me try to force again.

**Comment by tagyoureit** [comment]
@arrmo Hey bud!  Can you post a packet capture and I'll step through it.  I have an idea why it might be happening but would be useful to see your packets coming through.

**Comment by arrmo** [comment]
Hi @tagyoureit. Good to hear from you! Hope all is good there, and you are staying safe.

Absolutely, will capture this, NP at all. Any particular flags you want set (or debug levels)?

And thanks @rstrouse, really liking how all the pieces are working together! As @tagyoureit will confirm, I seem to find the very odd things ... LOL!

**Comment by rstrouse** [comment]
Not really odd just different than my setup.  I am running IntelliCenter and there are only a few differences.  One of them happens to be IntelliBrite control.  

However, I noticed that you have duplicate schedules.  Are there really 4 schedules defined at the panel or are we taking the effort to show you 2 schedules twice just in case you missed them the first time.

**Comment by arrmo** [comment]
Duplicate schedules? Sorry, you lost me there. The two pics are just vs. time, to show the Intellibrite "sneaking in" over time (i.e. it's not there right after refresh).

Or did I miss your point?

**Comment by rstrouse** [comment]
Oh that's funny.  One is AM and one is PM.  Totally missed that and probably need to pull out a bunch of those find the difference puzzles to make sure none of my breakers are tripped.

**Comment by arrmo** [comment]
LMAO - just say you're watching football, with adult beverages handy 😉

**Comment by arrmo** [comment]
FYI, just upgraded to the latest (master, and v6.5 of njsPC). Now I see some new features added? LOL! Tried removing them manually, but they reappear?
![image](https://user-images.githubusercontent.com/8756384/108637750-940d3a80-7451-11eb-9b02-ea80450ff78e.png)

Thanks!

**Comment by rstrouse** [comment]
@arrmo I would like you to include a include a replay.  Oddly it looks like the flags that normally indicate the feature is not assigned are actually set.  The fact that there isn't a name on them tells me that on your version of the firmware the feature circuit has a valid type on it but is probably not assigned to one of the internal names.  I just want to be sure of the incoming flags for the feature.

**Comment by arrmo** [comment]
Sure, NP! Let me dig into how to do a replay ... LOL!

**Comment by rstrouse** [comment]
Click on the hamburger menu and navigate to the logging tab.  Then press capture replay.  Leave the checkbox checked.

After the status changes to ready. Return to the logging tab and stop the capture.  It will download a zip to your browser.

**Comment by arrmo** [comment]
Very nice! File attached - just yell if you want me to try anything.

Thanks!
[replay.zip](https://github.com/rstrouse/nodejs-poolController-dashPanel/files/6039490/replay.zip)

**Comment by rstrouse** [comment]
Thanks for that.  I am talking with @tagyoureit about this.  I believe this comes down to these features being set up as generic but they don't have a name assigned to them.

**Comment by arrmo** [comment]
NP! Thanks for digging in to it. And no panic, that's for sure.

**Comment by rstrouse** [comment]
We have fixed the problem but it is only deployed in next so if you want to switch to the next repo you can do so using the following commands from the nodejs-PoolController directory.  If not the undefined features will go away in the next official release.
`sudo git checkout next`
`sudo git pull`

**Comment by arrmo** [comment]
Yep, that fixed it - thanks! I have been running next, so it was just a git pull (upstream next).

FYI, Intellibrite still "sneaks" in ... LOL!
